### DISCUSSION TITLE
Trying to send different format message to the same server and port

### DISCUSSION THREAD

**Discussion by daemondong** [syslog-ng/syslog-ng#4152] [Q&A] [answered]
# syslog-ng
## Platform is Linux engin-test 4.15.0-143-generic

I tried to send different format message to the same sever and port as below when I started the syslog-ng I got the error message.
Could any one tell me the reason ? Thank you.
By the way if I removed the "d_mongo_net" it worked well.

`
destination d_net {
    syslog("47.94.97.106" port(9002));
};

destination d_mongo_net {
\#    syslog("47.94.97.106" port(9002) template("__Mongo__ ${MESSAGE}"));
    syslog("47.94.97.106" port(9002) );
};
`

ERROR　Message:
`
shield@engin-test:~$ sudo systemctl restart syslog-ng.service
Job for syslog-ng.service failed because the control process exited with error code.
See "systemctl status syslog-ng.service" and "journalctl -xe" for details.
shield@engin-test:~$ sudo systemctl status syslog-ng.service
● syslog-ng.service - System Logger Daemon
   Loaded: loaded (/lib/systemd/system/syslog-ng.service; enabled; vendor preset: enabled)
   Active: failed (Result: exit-code) since Fri 2022-09-23 10:29:40 CST; 15s ago
     Docs: man:syslog-ng(8)
  Process: 27738 ExecReload=/bin/kill -HUP $MAINPID (code=exited, status=0/SUCCESS)
  Process: 13414 ExecStart=/usr/sbin/syslog-ng -F $SYSLOGNG_OPTS (code=exited, status=2)
 Main PID: 13414 (code=exited, status=2)
   Status: "Starting up... (Fri Sep 23 10:29:40 2022"

Sep 23 10:29:40 engin-test systemd[1]: syslog-ng.service: Service hold-off time over, scheduling restart.
Sep 23 10:29:40 engin-test systemd[1]: syslog-ng.service: Scheduled restart job, restart counter is at 5.
Sep 23 10:29:40 engin-test systemd[1]: Stopped System Logger Daemon.
Sep 23 10:29:40 engin-test systemd[1]: syslog-ng.service: Start request repeated too quickly.
Sep 23 10:29:40 engin-test systemd[1]: syslog-ng.service: Failed with result 'exit-code'.
Sep 23 10:29:40 engin-test systemd[1]: Failed to start System Logger Daemon.
shield@engin-test:~$ 
`

**Comment by gaborznagy** [comment, marked as answer]
Hi @daemondong !

I believe you have conflicting persist names in the configuration.

You should see a similar error in your logs:
`Automatic assignment of persist names failed, as conflicting persist-names were found. Please override the automatically assigned identifier using an explicit perist-name() option or remove the duplicated configuration elements;`

If this is the case, adding a `persist-name()` option to one of the destinations should help.

**Comment by daemondong** [comment]
Thank you very much. You are right. I added the persist names as below. It works now.
`
destination d_net {
    syslog("47.94.97.106" port(9002) persist-name("d_net"));
};

destination d_mongo_net {
    syslog("47.94.97.106" port(9002) template("__Mongo__ ${MESSAGE}") persist-name("d_mongo_net"));
};
`

**Comment by gaborznagy** [comment]
I'm glad it solved your problem. :+1:
